### DISCUSSION TITLE
Entrypoint Workflow issue

### DISCUSSION THREAD

**Discussion by mrraghavendra** [StackStorm/st2#5491] [General]
SUMMARY
We currently have 2.6.0 running in our STG environment, after some patch upgrade in Redhat Linux we see none of the workflows are running, keeps erroring out saying "Invalid input"

Installed:
![image](https://user-images.githubusercontent.com/37608050/145325334-ab309f7f-6f1e-4870-86fe-aa01ee6c50ac.png)

Error:
{
  "tasks": [
    {
      "state_info": "Failed to run task [error=Invalid input [name=st2.action, class=st2mistral.actions.stackstorm.St2Action, missing=['action_context']], wf=mwc.kafka_write_dev, task=write_message]:\nTraceback (most recent call last):\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/task_handler.py\", line 63, in run_task\n    task.run()\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/osprofiler/profiler.py\", line 160, in wrapper\n    result = f(*args, **kwargs)\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/tasks.py\", line 374, in run\n    self._run_new()\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/osprofiler/profiler.py\", line 160, in wrapper\n    result = f(*args, **kwargs)\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/tasks.py\", line 403, in _run_new\n    self._schedule_actions()\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/tasks.py\", line 475, in _schedule_actions\n    action.validate_input(input_dict)\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/actions.py\", line 322, in validate_input\n    self.action_def.action_class\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/utils.py\", line 66, in validate_input\n    raise exc.InputException(msg % tuple(msg_props))\nInputException: Invalid input [name=st2.action, class=st2mistral.actions.stackstorm.St2Action, missing=['action_context']]\n",
      "name": "write_message",
      "created_at": "2021-12-09 02:19:34",
      "updated_at": "2021-12-09 02:19:34",
      "id": "22e25809-6d50-4899-82cb-8a5d0be7ab0a",
      "workflow_execution_id": "3a89b725-c60e-4ad6-be92-d02fa41044f5",
      "state": "ERROR",
      "result": [],
      "published": {},
      "input": null,
      "workflow_name": "mwc.kafka_write_dev"
    }
  ],
  "result": "Failed to run task [error=Invalid input [name=st2.action, class=st2mistral.actions.stackstorm.St2Action, missing=['action_context']], wf=mwc.kafka_write_dev, task=write_message]:\nTraceback (most recent call last):\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/task_handler.py\", line 63, in run_task\n    task.run()\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/osprofiler/profiler.py\", line 160, in wrapper\n    result = f(*args, **kwargs)\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/tasks.py\", line 374, in run\n    self._run_new()\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/osprofiler/profiler.py\", line 160, in wrapper\n    result = f(*args, **kwargs)\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/tasks.py\", line 403, in _run_new\n    self._schedule_actions()\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/tasks.py\", line 475, in _schedule_actions\n    action.validate_input(input_dict)\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/actions.py\", line 322, in validate_input\n    self.action_def.action_class\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/utils.py\", line 66, in validate_input\n    raise exc.InputException(msg % tuple(msg_props))\nInputException: Invalid input [name=st2.action, class=st2mistral.actions.stackstorm.St2Action, missing=['action_context']]\n",
  "extra": {
    "state_info": "Failed to run task [error=Invalid input [name=st2.action, class=st2mistral.actions.stackstorm.St2Action, missing=['action_context']], wf=mwc.kafka_write_dev, task=write_message]:\nTraceback (most recent call last):\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/task_handler.py\", line 63, in run_task\n    task.run()\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/osprofiler/profiler.py\", line 160, in wrapper\n    result = f(*args, **kwargs)\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/tasks.py\", line 374, in run\n    self._run_new()\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/osprofiler/profiler.py\", line 160, in wrapper\n    result = f(*args, **kwargs)\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/tasks.py\", line 403, in _run_new\n    self._schedule_actions()\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/tasks.py\", line 475, in _schedule_actions\n    action.validate_input(input_dict)\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/actions.py\", line 322, in validate_input\n    self.action_def.action_class\n  File \"/opt/stackstorm/mistral/lib/python2.7/site-packages/mistral/engine/utils.py\", line 66, in validate_input\n    raise exc.InputException(msg % tuple(msg_props))\nInputException: Invalid input [name=st2.action, class=st2mistral.actions.stackstorm.St2Action, missing=['action_context']]\n",
    "state": "ERROR"
  }
}

ISSUE TYPE

**Comment by arm4b** [comment]
The latest StackStorm version is `v3.6.0`, Mistral workflow engine was deprecated a long time ago and StackStorm Enterprise is not existent anymore.

`v2.6.0` is from 2018 and I'm afraid we don't support that version. So it would be hard for someone to navigate where the problem is.
